packages/ERTP/src/paymentLedger.js

```diff
@@ -81,6 +86,7 @@ export const preparePaymentLedger = (
   optShutdownWithFailure = undefined,
 ) => {
   /** @type {Brand<K>} */
+  // @ts-expect-error XXX callWhen
```

Comment:
help me understand this one a bit?

(`ERTP/src/paymentLedger.js` is kinda critical. Before approving changes here, I need to grok deeply.)

Reply:
`callWhen` in an interface guard transforms the parameters of a function before calling it. It awaits all the erefs and then passes presences into the function instead of promises.

The reason for the XXX is that the TS defs can't handle this yet. There will have to be some inspection of the guards to infer which params have been awaited and which haven't.